internal/emulator/snapshot.go

```diff
@@ -0,0 +1,196 @@
+/*
+ * Flow CLI
+ *
+ * Copyright 2019 Dapper Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package emulator
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/onflow/flow-cli/internal/command"
+	"github.com/onflow/flow-cli/pkg/flowkit"
+	"github.com/onflow/flow-cli/pkg/flowkit/services"
+	"github.com/onflow/flow-cli/pkg/flowkit/util"
+
+	"github.com/spf13/cobra"
+)
+
+type SnapshotFlag struct {
+}
+
+var snapshotFlag = SnapshotFlag{}
+
+var SnapshotCmd = &command.Command{
+	Cmd: &cobra.Command{
+		Use:     "snapshot <snapshotName>",
+		Short:   "Save/Load a emulator snapshot",
+		Example: "flow emulator snapshot testSnapshot",
```

Comment:
does it make sense to separate into two actions, `flow emulator snapshot create|load {name}`.

Reply:
yeah I was thinking this, then we can also add `list`